controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationExtensionFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 JBoss by Red Hat.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.as.controller.persistence;
+
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.ServiceLoader;
+
+/**
+ *
+ * @author Emmanuel Hugonnet (c) 2021 Red Hat, Inc.
+ */
+public class ConfigurationExtensionFactory {
```

Comment:
Could you please provide some javadoc? At least specify that only one ConfigurationExtension will be loaded by the ServiceLoader...